sleap_roots/bases.py

```diff
@@ -25,8 +34,15 @@ def get_root_lengths(pts: np.ndarray) -> np.ndarray:
 
     Returns:
         Array of root lengths of shape (instances,).
+        If there is no root, or the roots is one point only (all of the rest of the
+        points are NaNs), an array of NaNs with shape (len(pts),) is returned.
+        This is also the case for non-contiguous points at the moment.
     """
     segment_diffs = np.diff(pts, axis=1)
     segment_lengths = np.linalg.norm(segment_diffs, axis=-1)
-    total_lengths = np.nansum(segment_lengths, axis=-1)
+
+    if np.isnan(segment_lengths).all():
+        total_lengths = np.empty((len(pts),)) ** np.nan
```

Comment:
Why are you exponentiating an empty array with `np.nan`? I think this is what you want:

```py
total_lengths = np.full((len(pts),), np.nan)
```

Reply:
Done.